s3/s3_filestore.go

```diff
@@ -50,7 +50,7 @@ func NewFilestore(ctx context.Context, endpoint, bucketName string, autoCreateBu
 		return nil, fmt.Errorf("checking if bucket %q exists: %w", bucketName, err)
 	}
 
-	if !bucketExists && !autoCreateBucket {
+	if !bucketExists && !s3Options.bucketAutoCreate {
```

Comment:
The idea is, that we don't call the client in the constructor if `BucketAutoCreate` is not set.
The filestore should always be constructed, even if e.g. MinIO is not reachable. So we only check and create the bucket if `BucketAutoCreate` is set.

Reply:
I think now everything is as it should